Program.fs

```diff
@@ -665,10 +665,172 @@ let private validateProject (library : LibraryWithNpmDeps) =
+        let actionsByPackage =
+            libActions
+            |> List.groupBy (function
+                | ResolveAction.Install(_, pkg, _, _) -> pkg
+                | ResolveAction.InstallDev(_, pkg,_,_) -> pkg
+                | ResolveAction.Uninstall(_, pkg, _) -> pkg
+                | ResolveAction.UninstallDev(_, pkg, _) -> pkg
+                | ResolveAction.UnableToResolve(_, pkg, _, _) -> pkg)
+
+        let currentLibrary = libraries |> List.find (fun lib -> lib.Name = library)
+
+        for (package, pkgActions) in actionsByPackage do
+            // using List.find because we can assume that the dependency was part of the current library
+            let requiredPackage = currentLibrary.NpmDependencies |> List.find (fun pkg -> pkg.Name = package)
+            logger.Information("{Library} requires npm package {Package}", library, package)
+            logger.Information("  | -- Required range {Range} found in project file", requiredPackage.Constraint |> Option.map string |> Option.defaultValue requiredPackage.RawVersion)
+
+            let installedPackage = installedPackages |> List.tryFind (fun pkg -> pkg.Name = package)
+            match installedPackage with
+            | None ->
+                logger.Information("  | -- Missing {package} in package.json", package)
+            | Some installed ->
+                match installed.Range, installed.Installed with
+                | Some range, Some version ->
+                    logger.Information("  | -- Used range {Range} in package.json", range.ToString())
+                    logger.Information("  | -- Found installed version {Version}", version.ToString())
+                | _ ->
+                    ignore()
+
+            let nodeCmd npm yarn =
+                match nodeManager with
+                | NodeManager.Npm -> npm
+                | NodeManager.Yarn -> yarn
+
+            // package actions one of the following
+            // - Install missing package
+            // - Uninstall from "devDependencies" and re-install into "dependencies"
+            // - Uninstall from "dependencies" and re-install "devDependencies"
+            // - Uninstall old (or "too new") package and re-install based on resolved version
+            // - Unable to resolve with errors
+            match pkgActions with
+            | [ ResolveAction.Install(_, _, version, range) ] ->
+                if range.Contains "&&"
+                then logger.Information("  | -- Required version constraint from multiple projects [{Range}]", package, range)
+                let installationCommand =
+                    nodeCmd
+                      (sprintf "npm install %s@%s --save" package version)
+                      (sprintf "yarn add %s@%s" package version)
+                logger.Information("  | -- Resolve manually using '{Command}'", installationCommand)
+
+            | [ ResolveAction.UninstallDev(_); ResolveAction.Install(_, _, version, range) ] ->
```

Comment:
Are we sure that `UninstallDev` and `Install` are always next to each other for the same library?

Reply:
Yes, that's how [I added them](https://github.com/Zaid-Ajaj/Femto/blob/master/Program.fs#L272-L275)